internal/webhook/kubernetes/warehouse/webhook.go

```diff
@@ -207,11 +207,25 @@ func (w *webhook) validateImageSub(
 	seen uniqueSubSet,
 ) field.ErrorList {
 	var errs field.ErrorList
-	if err := validateSemverConstraint(
-		f.Child("semverConstraint"),
-		sub.SemverConstraint,
-	); err != nil {
-		errs = field.ErrorList{err}
+	// Give precedence to the constraint field
+	// However, validate semVerConstraint field if
+	// semVer image strategy is used
+	if sub.ImageSelectionStrategy == kargoapi.ImageSelectionStrategySemVer {
+		if sub.SemverConstraint != "" {
+			if err := validateSemverConstraint(
+				f.Child("semverConstraint"),
+				sub.SemverConstraint,
+			); err != nil {
+				errs = field.ErrorList{err}
+			}
+		} else if sub.Constraint != "" {
+			if err := validateSemverConstraint(
+				f.Child("constraint"),
+				sub.Constraint,
+			); err != nil {
+				errs = field.ErrorList{err}
+			}
```

Comment:
```suggestion
		if sub.SemverConstraint != "" {
			if err := validateSemverConstraint(
				f.Child("semverConstraint"),
				sub.SemverConstraint,
			); err != nil {
				errs = field.ErrorList{err}
			}
		}

		if sub.Constraint != "" {
			if err := validateSemverConstraint(
				f.Child("constraint"),
				sub.Constraint,
			); err != nil {
				errs = field.ErrorList{err}
			}
```

Now that you have CEL in place, you can just check the value if it is set. No need to complicate this with `else if` type of things.

Additionally, I would: `errs = append(errs, err)` to prevent the list from being reset if logic is ever introduced above this block.